src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -9638,6 +9638,79 @@ public static String wrapIfMissing(final String str, final String wrapWith) {
         return builder.toString();
     }
 
+    /**
+     * Method that assembles all the numbers, form the passed string and returns them as list.
+     * It is important to note here, is that bu 'number' method assume any digit sequence, that
+     * can (but not necessary at all) contains dot within it (I mean just plain old floats,
+     * something like 51.82)
+     *
+     * For example, you may pass a string "21.2 days 3 minutes 22 seconds". For this particular string
+     * the result list of doubles will look like this : [21.2, 3.0, 22.0]
+     *
+     * if string contains invalid numbers (for example this string contains
+     * not valid number: "My height is 1234.23.13" This is invalid because it
+     * is not clear how to parse this part - 1234.23.13), {@link NumberFormatException}
+     * will be thrown. Though if string will contain number, where right
+     * after second dot resides not a number, or, any other char, then this
+     * case will be considered as valid. For example, this string contains
+     * only valid numbers: "My pulse is 90.123. and weight is 78.2"
+     * In this case sequence "90.123." will be considered as "90.123", as well as
+     * sequence "90." (imagine that there is no digit right after dot) will be
+     * considered as 90.0 double.
+     *
+     * @param stringThatContainsNumbers - string, that contains number or several numbers.
+     *                                 Not necessary integers, may be numbers with float point.
+     * @return - list of numbers, that this particular string contains
+     *
+     * @throws NumberFormatException - see documentation clarification about cases when thrown above
+     */
+    public static List<Double> extractNumbersFromString(String stringThatContainsNumbers) {
```

Comment:
The API name says Number but the return type says Double. They should match.

Reply:
So, ok, and what is exactly your proposition?) Btw all the types you mentioned extends Number class, that`s why from my point of view it is valid. Explain please more to me why it does not work